live/src/main/java/org/dbpedia/extraction/live/feeder/RCStreamFeeder.java

```diff
@@ -142,6 +146,38 @@ public void on(String event, IOAcknowledge ack, com.google.gson.JsonElement... a
      */
     @Override
     public void onError(SocketIOException socketIOException) {
-        logger.error("An error in the RCStream connection occured: " + socketIOException.getMessage());
+        if (reconnectAllowed()) {
+            logger.warn("An error in the RCStream connection occurred: " + socketIOException.getMessage() + " Trying to reconnect...");
+            try {
+                TimeUnit.SECONDS.sleep(10);
+                connect();
+            } catch (MalformedURLException exp) {
+                logger.error(ExceptionUtil.toString(exp), exp);
+            } catch(InterruptedException ex) {
+                Thread.currentThread().interrupt();
+            }
+        } else {
+            logger.error("An error in the RCStream connection occurred "
```

Comment:
This patch is a very good addition but Live is currently not so good with such failures. As a hack, whenever something fails I try to do a cleanup and restart  the service with 
```
Main.stopLive();
System.exit(1);
```

we are currently missing a way to actively monitor the status of the service